lib/3scale_toolbox/proxy_logger.rb

```diff
@@ -0,0 +1,25 @@
+module ThreeScaleToolbox
+  class ProxyLogger < BasicObject
+    instance_methods.each do |meth|
```

Comment:
There is no need to undefine those methods anymore no?

Reply:
There is no need to `undef`, right. Fixed.

Regarding Delegator/SimpleDelegator: I am trying to intercept calls, log input/output and measure elapsed time. This is done by proxy pattern. Decorator pattern using Delegator/SimpleDelegator is not meant for that, as long as I understood it.  Delegating is meant to add functionality in a class, but still behave like that. Besides solves inheritance issues to behave  in a way that's specific to delegated object, like `.class` method returned info.

Reply:
Delegator forwards all messages to the delegated objects. If you want to intercept all of them, then just need to override method_missing in the same way you do here, except instead of actually delegating the message sending just call super.

Delegator implements methods for equality, hashing, comparison, etc. that might not be needed in this case. The implementation is trivial, so it is not even about writing less code.

I just feel it would be a bit more correct because delegating all the methods to some other object is the purpose of the SimpleDelegator as mentioned in the docs:
> A concrete implementation of Delegator, this class provides the means to delegate all supported method calls to the object passed into the constructor